File: backend/src/test/java/lan/dk/podcastserver/manager/worker/TypeTest.java
package lan.dk.podcastserver.manager.worker;

import lan.dk.podcastserver.manager.worker.Type;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Created by kevin on 28/06/15 for Podcast Server
 */
public class TypeTest {

    @Test
    public void should_have_key_and_name() {
        /* Given */ Type type = new Type("Key", "Value");
        /* Then */
        assertThat(type.key()).isEqualTo("Key");
        assertThat(type.name()).isEqualTo("Value");
    }

}

File: backend/src/test/kotlin/com/github/davinkevin/podcastserver/update/updaters/TypeTest.kt
package com.github.davinkevin.podcastserver.update.updaters

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

/**
 * Created by kevin on 28/06/15 for Podcast Server
 */
class TypeTest {

    @Test
    fun should_have_key_and_name() {
        /* Given */
        /* When */
        val type = Type("Key", "Value")
        /* Then */
        assertThat(type.key).isEqualTo("Key")
        assertThat(type.name).isEqualTo("Value")
    }

    @Test
    fun `should be equal if has same key and name`() {
        /* Given */
        val k = "key"
        val n = "name"
        /* When */
        val t1 = Type(k, n)
        val t2 = Type(k, n)
        /* Then */
        assertThat(t1).isEqualTo(t2)
    }

}
